framework/base/Component.php

```diff
@@ -456,6 +463,13 @@ public function behaviors()
     public function hasEventHandlers($name)
     {
         $this->ensureBehaviors();
+
+        foreach ($this->_eventWildcards as $wildcard => $handlers) {
+            if (StringHelper::matchWildcard($wildcard, $name) && !empty($handlers)) {
```

Comment:
Condition could be reverted. empty check costs less.

Reply:
optimized